app/Http/Controllers/EventController.php

```diff
@@ -15,103 +15,117 @@ class EventController extends Controller
 {
     public function create(Request $request)
     {
-			    $this -> validate($request, [
-			        'name' => 'required|max:100',
-			        'description' => 'required|not_in:<br>',
-			        'subject' => 'required|not_in:0',
-			        'quizimage' => 'image|max:1000',
-			        'start_time' => 'required|after:Current_Date_Time',
-			        'end_time' => 'required|after:start_time',
-			        'duration' => 'required|numeric|not_in:0',
-			        'correct_mark' => 'required|numeric|not_in:0',
-			        'wrong_mark' => 'required|numeric',
-			        'display_ques' => 'required|numeric|not_in:0',
-			        'newsubject' => 'sometimes|not_in:subject',
-			    ]
-				,[
-					'not_in' => 'The :attribute field is required.'
-				]);
+	    $this -> validate($request,
+		    	[
+		        'name' => 'required|max:100',
+		        'description' => 'required|not_in:<br>',
+		        'subject' => 'required|not_in:0',
+		        'quizimage' => 'image|max:1000',
+		        'start_time' => 'required|after:Current_Date_Time',
+		        'end_time' => 'required|after:start_time',
+		        'duration' => 'required|numeric|not_in:0',
+		        'correct_mark' => 'required|numeric|not_in:0',
+		        'wrong_mark' => 'required|numeric',
+		        'display_ques' => 'required|numeric|not_in:0',
+		        'newsubject' => 'sometimes|not_in:subject',
+		    ]
+			,[
+				'not_in' => 'The :attribute field is required.'
+			]
+		);
+
+	    $task = new Event;
+	    $task->name = $request->name;
+	    $task->description = $request->description;
+	    if(!is_null($request->newsubject))
+	    {
+	    	//Add new Subject to subject table
+	    	$subject = new Subject;
+	    	$subject->name = $request->newsubject;
+	    	$subject->save();
+	    	$task->subid = $subject->id;
+	    }
+	    else
+	    {
+	    	Subject::findOrFail($request->subject);
```

Comment:
Are you handling the error ?

Reply:
findOrFail will throw Model Not Found exception so, are you showing any message to the user?